cai_causal_graph/causal_graph.py

```diff
@@ -1839,8 +1871,15 @@ def to_gml_string(self) -> str:
         return '\n'.join(networkx.generate_gml(self.to_networkx()))
 
     @classmethod
-    def from_dict(cls, d: dict) -> CausalGraph:
-        """Construct a `cai_causal_graph.causal_graph.CausalGraph` instance from a Python dictionary."""
+    def from_dict(cls, d: dict, validate: bool = True) -> CausalGraph:
+        """
+        Construct a `cai_causal_graph.causal_graph.CausalGraph` instance from a Python dictionary.
+
+        :param d: Dictionary to build a graph from.
+        :param validate: Whether to perform validation checks. The validation checks will raise if
+            any cycles are introduced to the graph by adding the edge. This should only be disabled to speed up
+            this method in situations where it is known that the serialized graph is acyclic. Default is `True`.
```

Comment:
A <-> B -> C is not really acyclic. Would say "... that the serialized graph is valid." instead